### PR TITLE
Add Pyformance metrics around forking behavior

### PR DESCRIPTION
**Pull request by dplumb94**

### CODE REVIEW

File: validator/sawtooth_validator/journal/publisher.py [hunk 1]
```diff
@@ -477,6 +479,11 @@ def __init__(self,
         self._pending_batch_gauge = \
             metrics_registry.gauge('pending_batch_gauge') \
             if metrics_registry else None
+        if metrics_registry:
+            self._blocks_published_count = CounterWrapper(
+                metrics_registry.counter('blocks_published_count'))
```
Comment (peterschwarz) [inline validator/sawtooth_validator/journal/publisher.py]: Is this really 'block_claimed_count'?

File: validator/sawtooth_validator/journal/back_pressure_handlers.py [hunk 2]
```diff
@@ -31,11 +34,21 @@ class ClientBatchSubmitBackpressureHandler(Handler):
...
 
-    def __init__(self, can_accept_fn, queue_info_fn):
+    def __init__(self, can_accept_fn, queue_info_fn, metrics_registry=None):
         self._can_accept = can_accept_fn
         self._queue_info = queue_info_fn
         self._applying_backpressure = False
 
+        if metrics_registry:
+            self._bp_batches_rejected_count = CounterWrapper(
+                metrics_registry.counter('bp_batches_rejected_count'))
```
Comment (vaporos) [inline validator/sawtooth_validator/journal/back_pressure_handlers.py]: The abbreviation is problematic here, as no one knows what bp means.  Does the longer string fail to work?

File: validator/sawtooth_validator/journal/chain.py [hunk 3]
```diff
@@ -13,6 +13,7 @@
 # limitations under the License.
 # ------------------------------------------------------------------------------
 
+# pylint: disable=too-many-lines
```
Comment (vaporos) [inline validator/sawtooth_validator/journal/chain.py]: This should really be disabled in pylintrc, not per-file.  This must have been re-enabled accidentally or due to changes in pylint.

File: validator/sawtooth_validator/journal/chain.py [hunk 4]
```diff
@@ -508,6 +516,10 @@ def run(self):
                  self._result["uncommitted_batches"]) = \
                     self._compute_batch_change(new_chain, cur_chain)
 
+                if new_chain[0].previous_block_id != \
+                        self._chain_head.identifier:
+                    self._fork_changes_count.inc()
```
Comment (vaporos) [inline validator/sawtooth_validator/journal/chain.py]: Consider "chain_head_moved_to_fork_count".

File: validator/sawtooth_validator/journal/chain.py [hunk 5]
```diff
@@ -747,6 +764,7 @@ def on_block_validated(self, commit_new_block, result):
         """
         try:
             with self._lock:
+                self._blocks_considered_count.inc()
```
Comment (vaporos) [inline validator/sawtooth_validator/journal/chain.py]: If this is the number of blocks that passed validation, consider "block_validated_count".  But really it seems like we should have metrics on the resulting status instead.

File: validator/sawtooth_validator/journal/back_pressure_handlers.py (hunk 2 above)
Comment (dplumb94) [reply validator/sawtooth_validator/journal/back_pressure_handlers.py]: It just required an ugly line break. I'll change it as the variable names are more important in this case.

Comment (dplumb94) [reply validator/sawtooth_validator/journal/back_pressure_handlers.py]: Also fails to pass lint as `backpressure_batches_rejected_count` is over 30 characters. Is `batches_rejected_count` too vague in this context?

File: validator/sawtooth_validator/journal/chain.py (hunk 5 above)
Comment (dplumb94) [reply validator/sawtooth_validator/journal/chain.py]: This is the number of blocks that have gone through validation but are not necessarily valid.

**Pull request merged**